gst-plugin-pravega/src/pravegasrc/imp.rs

```diff
@@ -161,8 +161,9 @@ impl Default for Settings {
 enum State {
     Stopped,
     Started {
-        reader: Arc<Mutex<CountingReader<BufReader<SeekableTake<ByteReader>>>>>,
-        index_searcher: Arc<Mutex<IndexSearcher<ByteReader>>>,
+        reader: Arc<Mutex<CountingReader<BufReader<SeekableTake<SyncByteReader>>>>>,
+        index_searcher: Arc<Mutex<IndexSearcher<SyncByteReader>>>,
+        client_factory: ClientFactory,
```

Comment:
Compile shows: 
```
warning: field is never read: `client_factory`
```

Reply:
add the comment at line 166. client_factory has to be saved to keep the tokio runtime.